AdvLoggerPkg/AdvLoggerOsConnectorPrm/AdvLoggerOsConnectorPrm.c

```diff
@@ -0,0 +1,162 @@
+/** @file AdvLoggerOsConnectorPrm.c
+
+  This driver gives an interface to OS components to fetch the AdvancedLogger memory log.
+
+  Copyright (c) Microsoft Corporation
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+**/
+
+#include <PrmModule.h>
+
+#include <Library/BaseLib.h>
+#include <Library/PrintLib.h>
+#include <Library/UefiLib.h>
+#include <Library/BaseMemoryLib.h>
+#include <AdvancedLoggerInternal.h>
+
+#include "AdvLoggerOsConnectorPrm.h"
+
+typedef struct {
+  VOID      *OutputBuffer;
+  UINT32    *OutputBufferSize;
+} ADVANCED_LOGGER_PRM_PARAMETER_BUFFER;
+
+/**
+    CheckAddress
+
+    The address of the ADVANCE_LOGGER_INFO block pointer is captured before END_OF_DXE.
+    LogBufferOffset, LogCurrentOffset, and LogBufferSize, could be written to by untrusted code.  Here, we check that
+    the offsets are within the allocated mLoggerInfo space, and that LogBufferSize, which is used in multiple places
+    to see if a new message will fit into the log buffer, is valid.
+
+    @param          DataBuf     PRM static data buffer containing LoggerInfo pointer
+
+    @return         BOOLEAN     TRUE - LoggerInfo passes security checks
+    @return         BOOLEAN     FALSE- LoggerInfo failed security checks
+
+**/
+STATIC
+BOOLEAN
+ValidateInfoBlock (
+  ADV_LOGGER_PRM_DATA_BUFFER  *DataBuf

```

Comment:
Might as well add the direction here.

```suggestion
  IN  ADV_LOGGER_PRM_DATA_BUFFER  *DataBuf
```